textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/source/IndexSearchSourceOperatorTest.java

```diff
@@ -0,0 +1,118 @@
+import org.apache.lucene.search.Query;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import edu.uci.ics.textdb.api.common.ITuple;
+import edu.uci.ics.textdb.api.storage.IDataReader;
+import edu.uci.ics.textdb.api.storage.IDataStore;
+import edu.uci.ics.textdb.api.storage.IDataWriter;
+import edu.uci.ics.textdb.common.constants.LuceneConstants;
+import edu.uci.ics.textdb.common.constants.TestConstants;
+import edu.uci.ics.textdb.common.exception.DataFlowException;
+import edu.uci.ics.textdb.storage.LuceneDataStore;
+import edu.uci.ics.textdb.storage.reader.LuceneDataReader;
+import edu.uci.ics.textdb.storage.writer.LuceneDataWriter;
+
+/**
+ * @author akshaybetala
+ *
+ */
+public class IndexSearchSourceOperatorTest {
+
+	private IDataWriter dataWriter;
+	private IndexSearchSourceOperator indexSearchSourceOperator;
+	private IDataStore dataStore;
+	private Analyzer analyzer;
+
+	@Before
+	public void setUp() throws Exception {
+		dataStore = new LuceneDataStore(LuceneConstants.INDEX_DIR, TestConstants.SCHEMA_PEOPLE);
+		analyzer = new StandardAnalyzer();
+		dataWriter = new LuceneDataWriter(dataStore, analyzer);
+		dataWriter.clearData();
+		dataWriter.writeData(TestConstants.getSamplePeopleTuples());
+	}
+
+	@After
+	public void cleanUp() throws Exception {
+		dataWriter.clearData();
+	}
+
+	public List<ITuple> getTupleCount(String query) throws DataFlowException, ParseException {
+		String defaultField = TestConstants.ATTRIBUTES_PEOPLE.get(0).getFieldName();
+		QueryParser queryParser = new QueryParser(defaultField, analyzer);
+		Query queryObject = queryParser.parse(query);
+		IDataReader dataReader = new LuceneDataReader(dataStore, queryObject);
+		indexSearchSourceOperator = new IndexSearchSourceOperator(dataReader);
+		indexSearchSourceOperator.open();
+
+		List<ITuple> results = new ArrayList<ITuple>();
+		ITuple nextTuple = null;
+		while ((nextTuple = indexSearchSourceOperator.getNextTuple()) != null) {
+			results.add(nextTuple);
+		}
+		indexSearchSourceOperator.close();
+		return results;
+	}
+
+	@Test
+	public void testTextSearcWithMultipleTokens() throws DataFlowException, ParseException {
+		List<ITuple> results = getTupleCount(TestConstants.DESCRIPTION + ":Tall,Brown");
+		int numTuples = results.size();
+		Assert.assertEquals(3, numTuples);
+
+		for (ITuple tuple : results) {
+			String value = (String) tuple.getField(TestConstants.DESCRIPTION).getValue();
+			Assert.assertTrue(value.toLowerCase().contains("tall") || value.toLowerCase().contains("brown"));
+		}
+	}
+
+	@Test
+	public void testTextSearchWithSingleToken() throws DataFlowException, ParseException {
+		List<ITuple> results = getTupleCount(TestConstants.DESCRIPTION + ":Tall");
+		int numTuples = results.size();
+		for (ITuple tuple : results) {
+			String value = (String) tuple.getField(TestConstants.DESCRIPTION).getValue();
+			Assert.assertTrue(value.toLowerCase().contains("tall"));
+		}
+

```

Comment:
Now I understand the logic, which is NOT complete.  It only verifies of each returned tuple has the keyword, but it doesn't check if they include all the correct results.

A better test case is to construct the number of expected results, and compare this set with the set of returned results.  These two sets should be exactly the same.